tbdex/sign.go

```diff
@@ -26,3 +30,39 @@ func Sign(digester Digester, bearerDID did.BearerDID) (string, error) {
 
 	return signature, nil
 }
+
+// Digest generates a SHA-256 hash of the canonicalized input payload.
+func Digest(payload interface{}) ([]byte, error) {
+	payloadBytes, err := json.Marshal(payload)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal payload: %w", err)
+	}
+
+	canonicalized, err := jcs.Transform(payloadBytes)
+	if err != nil {
+		return nil, fmt.Errorf("failed to canonicalize payload: %w", err)
+	}
+
+	hash := sha256.Sum256(canonicalized)
+
+	return hash[:], nil
+}
+
+// VerifySignature verifies the given signature and the signed payload
+func VerifySignature(digester Digester, signature string) error {
+	if signature == "" {
+		return errors.New("could not verify message signature because signature is empty")
```

Comment:
maybe `expected non-empty signature` since we're using this function for resources as well?